hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/type/NestedTypesDDLTest.java

```diff
@@ -58,123 +61,138 @@ public static void beforeClass() {
 
     @Test
     public void test_createIfNotExists() {
-        execute(format("CREATE TYPE FirstType OPTIONS ('format'='java','javaClass'='%s')", FirstType.class.getName()));
-        execute(format("CREATE TYPE IF NOT EXISTS FirstType OPTIONS ('format'='java','javaClass'='%s')", SecondType.class.getName()));
+        newJavaType("FirstType", FirstType.class).create();
+        newJavaType("FirstType", SecondType.class).createIfNotExists();
 
         assertEquals(FirstType.class.getName(), storage.getType("FirstType").getJavaClassName());
     }
 
     @Test
     public void test_showTypes() {
-        execute(format("CREATE TYPE FirstType OPTIONS ('format'='java','javaClass'='%s')", FirstType.class.getName()));
-        execute(format("CREATE TYPE SecondType OPTIONS ('format'='java','javaClass'='%s')", SecondType.class.getName()));
+        newJavaType("FirstType", FirstType.class).create();
+        newJavaType("SecondType", SecondType.class).create();
+
         assertRowsAnyOrder("SHOW TYPES", rows(1, "FirstType", "SecondType"));
     }
 
     @Test
     public void test_dropNonexistentType() {
         assertThatThrownBy(() -> execute("DROP TYPE Foo"))
-                .isInstanceOf(HazelcastSqlException.class)
                 .hasMessage("Type does not exist: Foo");
 
         execute("DROP TYPE IF EXISTS Foo");
     }
 
     @Test
     public void test_createTwoTypesForSameJavaClass() {
-        execute(format("CREATE TYPE FirstType OPTIONS ('format'='java','javaClass'='%s')", FirstType.class.getName()));
-        execute(format("CREATE TYPE SecondType OPTIONS ('format'='java','javaClass'='%s')", FirstType.class.getName()));
+        newJavaType("FirstType", FirstType.class).create();
+        assertThatCode(() -> newJavaType("SecondType", FirstType.class).create())
+                .doesNotThrowAnyException();
     }
 
     @Test
     public void test_createTwoTypesForSamePortableClass() {
-        execute("CREATE TYPE FirstType(a INT, b INT) OPTIONS ('format'='portable','portableFactoryId'='123','portableClassId'='456')");
-        execute("CREATE TYPE SecondType(c VARCHAR, d VARCHAR) OPTIONS ('format'='portable','portableFactoryId'='123','portableClassId'='456')");
-
-        try (SqlResult result = instance().getSql().execute("CREATE OR REPLACE MAPPING " + "m(" +
-                "e varchar" +
-                ")" + " TYPE " + IMapSqlConnector.TYPE_NAME + " "
-                + "OPTIONS ("
-                + '\'' + OPTION_KEY_FORMAT + "'='" + JAVA_FORMAT + '\''
-                + ", '" + OPTION_KEY_CLASS + "'='" + Long.class.getName() + '\''
-                + ", '" + OPTION_VALUE_FORMAT + "'='" + PORTABLE_FORMAT + '\''
-                + ", '" + OPTION_VALUE_FACTORY_ID + "'='" + 123 + '\''
-                + ", '" + OPTION_VALUE_CLASS_ID + "'='" + 456 + '\''
-                + ", '" + OPTION_VALUE_CLASS_VERSION + "'='" + 0 + '\''
-                + ")"
-        )) {
-            assertThat(result.updateCount()).isEqualTo(0);
```

Comment:
this assert was not very valuable but still it checked some aspect of the behavior

Reply:
you might move it to `createOrReplace()` etc.

Reply:
The update count of a DDL being 0 should definitely be tested, but should it be tested whenever a DDL is issued?

Reply:
probably not